packages/shared/ReactFeatureFlags.js

```diff
@@ -91,10 +91,19 @@ export const enableTrustedTypesIntegration = false;
 // Flag to turn event.target and event.currentTarget in ReactNative from a reactTag to a component instance
 export const enableNativeTargetAsInstance = false;
 
+// Controls sequence of passive effect destroy and create functions.
+// If this flag is off, destroy and create functions may be interleaved.
+// When the falg is on, all destroy functions will be run (for all fibers)
+// before any create functions are run, similar to how layout effects work.
+// This flag provides a killswitch if that proves to break existing code somehow.
+export const runAllPassiveEffectDestroysBeforeCreates = false;
+
 // Controls behavior of deferred effect destroy functions during unmount.
 // Previously these functions were run during commit (along with layout effects).
 // Ideally we should delay these until after commit for performance reasons.
 // This flag provides a killswitch if that proves to break existing code somehow.
+//
+// WARNING Only enable this flag in conjunction with runAllPassiveEffectDestroysBeforeCreates.
```

Comment:
This constraint is a little gross ☹️ but it keeps the branching complexity low.

Reply:
Instead of two booleans you could use an enum, to prevent the invalid combination

Reply:
True. I considered that but our precedent has always been booleans for flags.

I don't feel strongly about this though.

Reply:
Alternative, in the code, you could also check for both booleans before you enable the "defer unmounts" feature.

Just want to make it harder for someone to accidentally toggle the wrong combination of GKs.

Reply:
Yeah, I hear you. That seems reasonable.